backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -44,38 +48,35 @@
   private static final String JSON_SECOND_PARTICIPANT_USERNAME = "secondParticipantUsername";
   private static final String JSON_DURATION = "duration";
 
+  // Create GmailFactory and use it to instantiate Email Notifier
+  private final GmailFactory gmFactory = new GmailFactory();
+  // Get DatastoreService and instiate Match and Participant Datastores
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final MatchDatastore matchDatastore = new MatchDatastore(datastore);
+  private final ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+  // Get participant username
+  private final UserService userService = UserServiceFactory.getUserService();
+  private EmailNotifier emailNotifier;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    System.out.println("Request received");
 
-    // Get participant username
-    UserService userService = UserServiceFactory.getUserService();
-    String email = userService.getCurrentUser().getEmail();
-    if (email == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
-      return;
-    }
-    String username = email.split("@")[0];
-
-    // Get DatastoreService and instiate Match and Participant Datastores
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    MatchDatastore matchDatastore = new MatchDatastore(datastore);
-    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
```

Comment:
We should not remove these lines.

Reply:
I think Jordan added these datastore instantiations as fields in this servlet now, so I think these might be fine to remove

Reply:
I didn't remove them i  made them global

Reply:
check line 54